pkg/sentry/fsimpl/fuse/fusefs.go

```diff
@@ -226,3 +240,46 @@ func (i *Inode) Open(ctx context.Context, rp *vfs.ResolvingPath, vfsd *vfs.Dentr
 	}
 	return fd.VFSFileDescription(), nil
 }
+
+// NewNode implements kernfs.Inode.NewNode.
+func (i *Inode) NewNode(ctx context.Context, name string, opts vfs.MknodOptions) (*vfs.Dentry, error) {
+	kernelTask := kernel.TaskFromContext(ctx)
+	if kernelTask == nil {
+		log.Warningf("fusefs.Inode.NewNode: couldn't get kernel task from context", i.NodeID)
+		return nil, syserror.EINVAL
+	}
+	in := linux.FUSEMknodReq{
+		MknodIn: linux.FUSEMknodIn{
+			Mode:  uint32(opts.Mode),
+			Rdev:  (opts.DevMinor & 0xff) | (opts.DevMajor << 8) | (opts.DevMinor & ^uint32((0xff)) << 12),
+			Umask: uint32(kernelTask.FSContext().Umask()),
+		},
+		Name: name,
+	}
+	req, err := i.fs.conn.NewRequest(auth.CredentialsFromContext(ctx), uint32(kernelTask.ThreadID()), i.NodeID, linux.FUSE_MKNOD, &in)
+	if err != nil {
+		return nil, err
+	}
+	return i.newEntry(kernelTask, req, name, opts.Mode.FileType())
+}
+
+// newEntry create new entry according to response from FUSE server, shared by FUSE_MKNOD, FUSE_MKDIR, FUSE_SYMLINK and FUSE_LINK.
```

Comment:
It might be more clear to document that this function also actually sends the request